### PR TITLE
Bugfix/issue 126 icu vulnerability

### PR DESCRIPTION
**Pull request by aabhasg**
Upgraded icu4j version and update license header in multiple files.

### CODE REVIEW

File: unicode/build.gradle [hunk 1]
```diff
@@ -12,6 +12,6 @@ repositories {
 dependencies {
   compile 'com.google.googlejavaformat:google-java-format:1.0'
   compile 'com.squareup:javapoet:1.12.1'
-  compile 'com.ibm.icu:icu4j:4.8.2'
+  compile 'com.ibm.icu:icu4j:68.2'
```
Comment (aabhasg) [inline unicode/build.gradle:15]: It is the latest version available in maven repository.

**Comment by sjamesr** [comment]
The icu4j dependency exists only to generate unicode tables. You should rerun the unicode table generator and run the tests, I suspect they will fail.

**Comment by aabhasg** [comment]
Tried to rerun the tests as mentioned. All of them are running except few which are still failing on master. 
image

**Comment by aabhasg** [comment]
I tried rerunning test as mentioned. We are good, all the tests ran successfully.

**Comment by sjamesr** [comment]
You didn't update the UnicodeTables class,

```
./gradlew -q  :unicode:run > java/com/google/re2j/UnicodeTables.java
```

This causes test failures. The solution will be to remove the dependency on icu4j altogether. For this, we need to generate the unicode tables at runtime on-demand.

**Pull request closed without merging**